libraries/Wire/Wire.h

```diff
@@ -28,24 +28,23 @@
 #include "Stream.h"
 
 
-
 #ifndef I2C_BUFFER_LENGTH
+// DEPRECATED: Do not use BUFFER_LENGTH, prefer I2C_BUFFER_LENGTH
```

Comment:
I expect this may be a bit controversial. My idea is to use the new `I2C_BUFFER_LENGTH` functionality as a carrot to convince people to switch to the new define.